File: frontend/src/types/pin.ts
export interface Pin {
    id: string;
    messageId: string;
    chatId: string;
    content: string;
    userId: string;
    createdAt: Date;
    updatedAt?: Date;
}

export interface PinCreate {
    messageId: string;
    chatId: string;
    content: string;
    userId: string;
} 

File: backend/app/models/pin.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class PinBase(BaseModel):
    message_id: str
    chat_id: str
    content: str
    user_id: str

class PinCreate(PinBase):
    pass

class Pin(PinBase):
    id: str
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True 